### ISSUE TITLE
SMTP-Mail notifications have no Message-Id

### ISSUE THREAD

**Issue by Niduroki** [containrrr/shoutrrr#229]

**Comment by Niduroki** [comment]
rspamd (which my mail server uses) seems to take this way more (i.e. 25 times more) seriously, by default …

![Screenshot_20220311_114504](https://user-images.githubusercontent.com/1410802/157852887-471b1f94-2d79-4973-abbb-aab31f7262a7.jpeg)

Yeah, it does not increase authenticity, and spammers could definitely just add an UUID, too, a surprising amount of spammers do not do that, however.

-----

Do you have any source on the relaying server being supposed to insert the MID? My mail server does not seem to do that, and solutions like roundcube ([code-ref on GH](https://github.com/roundcube/roundcubemail/blob/master/program/include/rcmail_sendmail.php#L177) – [MID-generator function](https://github.com/roundcube/roundcubemail/blob/master/program/lib/Roundcube/rcube.php#L1680)) seem to insert their own MIDs, too.

In case you are really worried about uuids colliding you could append a unixtime (and/or some more random-characters) after the UUID-string, I guess.

**Comment by piksel** [comment]
Yeah, it's a somewhat tricky subject, since the RFC suggests that the `host` should generate the ID, using it's hostname (which is presumed to be uniquely pointing to this one host) as the suffix. Some e-mail clients generate ID's and put the host as `localhost` or the local computer name. Using the sending MTA's domain is probably not a good idea, since spam filters "know" what format different mail hosts use and put high scores on mismatching IDs.
Putting `UUID@<docker-host>` could be implemented though, and would have a pretty slim chance of collisions, even if a weak pseudo-random source is used.

**Comment by 57thStIncident** [comment]
This appears to be an issue for me -- trying to e-mail a google workspace address, it now seems to reject messages without Message-ID out of hand.

`relay=aspmx.l.google.com[142.251.179
.27]:25, delay=0.42, delays=0/0/0.12/0.29, dsn=5.7.1, status=bounced (host aspmx.l.google.com[142.251.179.27] said: 550-5.7.
1 [100.1.25.178] Messages missing a valid Message-ID header are not 550-5.7.1 accepted. For more information, go to 550-5.7.
1  https://support.google.com/mail/?p=RfcMessageNonCompliant and review 550 5.7.1 RFC 5322 specifications. 6a1803df08f44-880
82aa90d7si14807396d6.289 - gsmtp (in reply to end of DATA command))`

I don't know any way to work around this and make shoutrrr smtp to google workspace e-mail usable.

**Comment by piksel** [comment]
I haven't had time to implement this, but I could take a look at a PR if anyone is willing to do the implementation.

**Comment by atighineanu** [comment]
I could have a look as well. I assume a header check in a mock call should be enough to test it, correct?
